### DISCUSSION TITLE
adding Google Analytics failed

### DISCUSSION THREAD

**Discussion by JoeShi** [imfing/hextra#99] [Q&A]
Hi, 

I have followed the [Guide](https://imfing.github.io/hextra/docs/guide/configuration/#google-analytics) and updated my `hugo.yaml` file. But I cannot find the API invocation to google analytics.

How to reproduce
1. checkout this repo
2. `cd` into exampleSite folder
3. add `googleAnalytics: G-xxxx` in the `hugo.yaml` file
4. run `hugo server --logLevel debug --disableFastRender`

**Comment by imfing** [comment]
Hi, thank you for posting. Google Analytics will only be enabled for production, i.e. running `hugo server` locally won't enable the Google Analytics tag.
If you'd like to see the analytics tag in action, please run `hugo` to build the site and inspect the built HTML files.

I will update the docs for clarification.

**Comment by JoeShi** [comment]
Thank you for the explanation. I have run the hugo command to build the site and deploy on my webserver. However, I only saw it will send metrics to GA once when the site load the home page. It will not send metrics when I navigate to other paths, such as, `/docs`, `/contact`. Do you know how to track this?

**Reply by imfing** [reply]
It should track the paths automatically. The Google Analytics is added on the top level so it should be enabled for every page in your site.
Could you double check in your Google Analytics see if you could see metrics from the "Reports" tab?

Do you mind share your site repo if it's public? and I could check if any thing goes wrong.

**Reply by JoeShi** [reply]
Sorry, this is a private repo. What I observed is it cannot load GA any more. Since I want to using the home page of your site. So what I did is copy the following files into the `layouts` folder.

- partials/utils/page-width.html
- hextra-home.html
- index.html

Not sure if it is caused by this?

**Comment by JoeShi** [comment]
Sorry, this is a private repo. What I observed is it cannot load GA any more. Since I want to using the home page of your site. So what I did is copying the following files into the `layouts` folder, and all the files in content folder.

- partials/utils/page-width.html
- hextra-home.html
- index.html

Not sure if it is caused by this?

**Reply by imfing** [reply]
I don't think overriding the above files would cause any problem regarding Google Analytics.

What I recommend is to check your site by opening browser developer tool, and inspect the `head` element for the presence of the GA tags and scripts. Also double check the console to see if any errors occurred. 
Note that you need to disable ad blockers since they may affect the analytics.

**Reply by JoeShi** [reply]
Thank you. I did not find the GA tags and scripts in the `head` element. I just followed https://imfing.github.io/hextra/docs/guide/configuration/#google-analytics and add the GA configuration in the `hugo.yaml` file. Should I do something else?

I noticed that the Hugo documentation said there is an internal template https://gohugo.io/templates/internal/#google-analytics. Should I place it somewhere?

**Reply by imfing** [reply]
> Thank you. I did not find the GA tags and scripts in the head element. I just followed https://imfing.github.io/hextra/docs/guide/configuration/#google-analytics and add the GA configuration in the hugo.yaml file. Should I do something else?

I wonder which version of the theme are you using? 

If the config was set, then every page should include it after you run `hugo`

![image](https://github.com/imfing/hextra/assets/5097752/1a66c28a-8d29-43e3-9af4-94d53726ac8f)

> I noticed that the Hugo documentation said there is an internal template https://gohugo.io/templates/internal/#google-analytics. Should I place it somewhere?

No, the internal template is not used by Hextra. Instead, we shipped our own: https://github.com/imfing/hextra/blob/main/layouts/partials/google-analytics.html
